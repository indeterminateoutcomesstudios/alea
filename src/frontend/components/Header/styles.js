// @flow
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import { fontSize } from 'frontend/styles/common'

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  height: 4rem;
  line-height: 4rem;
  color: white;
  background-color: ${props => props.theme.backgroundInverted};
`

export const Column = styled.div`
  display: flex;
  flex-direction: row;
  flex: 1;
  justify-content: ${props =>
    props.right ? 'flex-end' : props.middle ? 'center' : 'flex-start'};
`

export const Button = styled(Link)`
  text-decoration: none;
  height: 4rem;
  min-width: 4rem;
  padding: 0 1rem;
  text-align: center;
  font-size: ${fontSize.normal};
  cursor: pointer;
  color: ${props => props.theme.colorInverted};

  &:hover {
    background: ${props => props.theme.backgroundInvertedSecondary};
  }
`

export const LogoWrapper = styled.div`
  padding-top: 12px;
  line-height: 1;
`

export const Avatar = styled.img`
  float: right;
  width: 3rem;
  height: 3rem;
  margin: 0.5rem 0 0.5rem 0.8rem;
  border-radius: 3px;
`
