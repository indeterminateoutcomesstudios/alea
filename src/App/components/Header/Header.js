/* @flow */
import React from 'react'
import Home from './Home'
import styles from './style.css'

type Props = {
  children?: Object
}

class Header extends React.Component<*, Props, *> {
  render () {
    const { children } = this.props

    return (
      <div className={styles.header}>
        <Home />
        { children }
      </div>
    )
  }
}

export default Header
