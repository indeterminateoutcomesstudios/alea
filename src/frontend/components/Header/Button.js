// @flow
import styled from 'styled-components'
import { fontSize } from 'frontend/styles/common'

const Button = styled.div`
  height: 4rem;
  width: ${props => (props.square ? '4rem' : 'auto')};
  padding: 0 1rem;
  text-align: center;
  font-size: ${fontSize.normal};
  cursor: pointer;
  color: ${props => props.theme.colorInverted};

  &:hover {
    background: ${props => props.theme.backgroundInvertedSecondary};
  }
`

export default Button
