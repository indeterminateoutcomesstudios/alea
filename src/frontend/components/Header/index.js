// @flow
import * as React from 'react'
import LogoSvg from 'frontend/components/Logo'
import slug from 'slugg'
import { Button, LogoWrapper, Container, Column } from './styles'
import type { DBProfile } from 'common/types'

type Props =
  | {
      isLoading: true,
      currentUserProfileQuery: null
    }
  | {
      isLoading: false,
      currentUserProfileQuery: {
        currentUser: { profile: DBProfile }
      }
    }

const Header = (props: Props) => {
  const currentUser = props.isLoading
    ? null
    : props.currentUserProfileQuery.currentUser

  return (
    <Container>
      <Column>
        <Button to="/sessions">Games</Button>
      </Column>
      <Column middle>
        <Button to="/">
          <LogoWrapper>
            <LogoSvg height="16px" />
          </LogoWrapper>
        </Button>
      </Column>
      <Column right>
        {currentUser && (
          <React.Fragment>
            <Button
              to={`/u/${slug(currentUser.profile.username)}/${
                currentUser.profile.id
              }`}
            >
              {currentUser.profile.username}
            </Button>
            <Button to="/settings">
              <i className="fa fa-cog" />
            </Button>
          </React.Fragment>
        )}
      </Column>
    </Container>
  )
}

export default Header
