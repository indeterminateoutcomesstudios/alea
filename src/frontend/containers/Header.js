// @flow
import { connect } from 'react-redux'
import { compose } from 'recompose'
import Header from 'frontend/components/Header'
// import queryHandler from 'frontend/components/queryHandler'
// import { getCurrentUserProfile } from 'frontend/graphql/queries/currentUser/getCurrentUserProfile'
import type { State } from 'frontend/store'

const mapStateToProps = (state: State) => ({
  userIsLoggedIn: state.currentUser.id !== null
})

export default compose(
  connect(mapStateToProps, {})
  // getCurrentUserProfile,
  // queryHandler({
  // queries: ['currentUserProfileQuery']
  // })
)(Header)
