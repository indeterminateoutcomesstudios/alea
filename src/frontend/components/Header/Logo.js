// @flow
import React from 'react'
import styled from 'styled-components'
import { Link } from 'react-router-dom'
import Button from './Button'
import LogoSvg from 'frontend/components/Logo'

const Wrapper = styled.div`
  margin: auto;
  display: flex;
`

const LogoButton = () => (
  <Wrapper>
    <Link to="/">
      <Button>
        <LogoSvg height="12px" />
      </Button>
    </Link>
  </Wrapper>
)

export default LogoButton
