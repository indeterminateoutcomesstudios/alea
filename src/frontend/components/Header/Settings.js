// @flow
import React from 'react'
import Button from './Button'
import { Link } from 'react-router-dom'

const Settings = () => {
  return (
    <Link to="/settings">
      <Button square>
        <i className="fa fa-cog" />
      </Button>
    </Link>
  )
}

export default Settings
