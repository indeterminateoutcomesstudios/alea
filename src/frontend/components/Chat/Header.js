// @flow
import React from 'react'
import styled from 'styled-components'
import Tooltip from '../Tooltip'
import { fontSize } from 'frontend/styles/common'
import { PinIcon, UnpinIcon } from 'frontend/components/icon'

const Container = styled.div`
  background-color: ${props =>
    props.isPinned ? props.theme.background : props.theme.backgroundInverted};
  display: flex;
  flex-direction: row;
`

const Toggle = styled.div`
  display: block;
  font-size: ${fontSize.small};
  color: ${props =>
    props.isPinned ? props.theme.color : props.theme.colorInverted};
  margin-right: 1rem;
  padding-left: 1rem;
  cursor: pointer;

  i {
    transform: ${props => (props.isPinned ? 'none' : 'rotate(45deg)')};
  }
`

const Outer = styled.div`
  margin-left: auto;
`

type Props = {
  isPinned: boolean,
  setChatPinned: Function
}

const Header = (props: Props) => {
  const { isPinned, setChatPinned } = props

  const tooltip = isPinned ? 'Unpin Chat' : 'Pin Chat'

  return (
    <Container isPinned={isPinned}>
      <Outer>
        <Tooltip placement="left" content={tooltip}>
          <Toggle onClick={() => setChatPinned(!isPinned)} isPinned={isPinned}>
            {isPinned ? <UnpinIcon /> : <PinIcon />}
          </Toggle>
        </Tooltip>
      </Outer>
    </Container>
  )
}

export default Header
