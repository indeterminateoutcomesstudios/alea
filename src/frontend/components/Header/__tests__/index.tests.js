// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import Header from '../index.js'

describe('Header component', () => {
  it('renders correctly when loading', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header
            isLoading={true}
            hasError={false}
            currentUserProfileQuery={{
              currentUser: { profile: { username: '' } }
            }}
          />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when logged in', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Header
            isLoading={false}
            hasError={false}
            currentUserProfileQuery={{
              currentUser: { profile: { username: 'test' } }
            }}
          />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
