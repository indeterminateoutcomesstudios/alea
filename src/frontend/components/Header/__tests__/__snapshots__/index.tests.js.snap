// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly when logged in 1`] = `
.c2 {
  height: 3.25rem;
  width: 3.25rem;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
}

.c4 {
  height: 3.25rem;
  width: auto;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
}

.c5 svg {
  height: 12px;
  fill: currentColor;
}

.c3 {
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  height: 3.25rem;
  width: auto;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
  font-size: 1.3rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 4.25rem;
  line-height: 4.25rem;
  color: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div>
        <a
          href="/sessions"
          onClick={[Function]}
        >
          <div
            className="c2"
          >
            <i
              className="fa fa-home"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c3"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="c4"
        >
          <div
            className="c5"
            dangerouslySetInnerHTML={
              Object {
                "__html": Object {},
              }
            }
            height="12px"
          />
        </div>
      </a>
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      className="c7"
    >
      
    </div>
    <a
      href="/settings"
      onClick={[Function]}
    >
      <div
        className="c2"
      >
        <i
          className="fa fa-cog"
        />
      </div>
    </a>
  </div>
</div>
`;

exports[`Header component renders correctly with no user 1`] = `
.c2 {
  height: 3.25rem;
  width: 3.25rem;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
}

.c4 {
  height: 3.25rem;
  width: auto;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
}

.c5 svg {
  height: 12px;
  fill: currentColor;
}

.c3 {
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c7 {
  height: 3.25rem;
  width: auto;
  line-height: 3.25rem;
  padding: 0 1rem;
  text-align: center;
  font-size: 1.5rem;
  cursor: pointer;
  border-radius: 5px;
  font-size: 1.3rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  height: 4.25rem;
  line-height: 4.25rem;
  color: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.5rem;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      onMouseOut={[Function]}
      onMouseOver={[Function]}
    >
      <div>
        <a
          href="/sessions"
          onClick={[Function]}
        >
          <div
            className="c2"
          >
            <i
              className="fa fa-home"
            />
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c3"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <div
          className="c4"
        >
          <div
            className="c5"
            dangerouslySetInnerHTML={
              Object {
                "__html": Object {},
              }
            }
            height="12px"
          />
        </div>
      </a>
    </div>
  </div>
  <div
    className="c6"
  >
    <div
      className="c7"
    >
      
    </div>
    <a
      href="/settings"
      onClick={[Function]}
    >
      <div
        className="c2"
      >
        <i
          className="fa fa-cog"
        />
      </div>
    </a>
  </div>
</div>
`;
